File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizCountryCodeSheet.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Search
import androidx.compose.material3.Icon
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@Composable
fun BizCountryCodeSheet(
    isVisible: Boolean,
    selectedCode: String,
    onDismissRequest: () -> Unit,
    onChange: (String) -> Unit
) {
    data class Country(val code: String, val name: String, val dialCode: String)
    val popularCountries = listOf(
        Country("KH", "Cambodia", "+855"),
        Country("KR", "Republic of Korea", "+82"),
        Country("CN", "China", "+86"),
        Country("JP", "Japan", "+81")
    )
    val allCountries = popularCountries + listOf(
        Country("US", "United States", "+1"),
        Country("GB", "United Kingdom", "+44"),
        Country("TH", "Thailand", "+66"),
        Country("VN", "Vietnam", "+84"),
        Country("SG", "Singapore", "+65")
    )

    var searchQuery by remember { mutableStateOf("") }
    var activeTab by remember { mutableStateOf("popular") }

    val filteredCountries = remember(activeTab, searchQuery) {
        val list = if (activeTab == "popular") popularCountries else allCountries
        if (searchQuery.isEmpty()) list
        else {
            val query = searchQuery.lowercase()
            list.filter { it.name.lowercase().contains(query) || it.dialCode.contains(query) }
        }
    }

    BizBottomSheet(
        isVisible = isVisible,
        onDismissRequest = onDismissRequest
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp)
                .padding(bottom = 32.dp)
                .heightIn(max = 600.dp)
        ) {
            Text(
                text = "Select Your Country Code",
                fontSize = 16.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF1A2A5E),
                modifier = Modifier.padding(bottom = 20.dp)
            )

            // Search Box (simplified placeholder for BizInput)
            OutlinedTextField(
                value = searchQuery,
                onValueChange = { searchQuery = it },
                placeholder = { Text("Search", color = Color(0xFF999999)) },
                leadingIcon = {
                    Icon(
                        imageVector = Icons.Default.Search,
                        contentDescription = "Search",
                        tint = Color(0xFF999999),
                        modifier = Modifier.size(18.dp)
                    )
                },
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 16.dp),
                shape = RoundedCornerShape(12.dp),
                colors = OutlinedTextFieldDefaults.colors(
                    unfocusedBorderColor = Color(0xFFEAEAEA),
                    focusedBorderColor = Color(0xFF0066CC)
                ),
                singleLine = true
            )

            // Segmented Button (simplified)
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 16.dp)
                    .background(Color(0xFFF5F5F5), RoundedCornerShape(8.dp))
                    .padding(4.dp)
            ) {
                val tabs = listOf("popular" to "Popular", "all" to "All Country")
                tabs.forEach { (value, label) ->
                    val isActive = activeTab == value
                    Box(
                        modifier = Modifier
                            .weight(1f)
                            .background(
                                color = if (isActive) Color.White else Color.Transparent,
                                shape = RoundedCornerShape(6.dp)
                            )
                            .clickable { activeTab = value }
                            .padding(vertical = 8.dp),
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = label,
                            fontSize = 14.sp,
                            fontWeight = if (isActive) FontWeight.SemiBold else FontWeight.Normal,
                            color = if (isActive) Color(0xFF1A2A5E) else Color(0xFF777777)
                        )
                    }
                }
            }

            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .verticalScroll(rememberScrollState()),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                filteredCountries.forEach { country ->
                    val isActive = selectedCode == country.dialCode
                    Surface(
                        shape = RoundedCornerShape(12.dp),
                        color = Color.White,
                        border = BorderStroke(
                            width = 1.5.dp,
                            color = if (isActive) Color(0xFF0066CC) else Color(0xFFEAEAEA)
                        ),
                        modifier = Modifier
                            .fillMaxWidth()
                            .clickable {
                                onChange(country.dialCode)
                                CoroutineScope(Dispatchers.Main).launch {
                                    delay(200)
                                    onDismissRequest()
                                }
                            }
                    ) {
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = 16.dp, vertical = 14.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Text(
                                text = "${country.name} (${country.dialCode})",
                                fontSize = 14.sp,
                                fontWeight = FontWeight.Medium,
                                color = if (isActive) Color(0xFF0066CC) else Color(0xFF111111)
                            )

                            Box(
                                modifier = Modifier
                                    .size(20.dp)
                                    .background(Color.Transparent, CircleShape),
                                contentAlignment = Alignment.Center
                            ) {
                                Surface(
                                    modifier = Modifier.fillMaxSize(),
                                    shape = CircleShape,
                                    color = Color.Transparent,
                                    border = BorderStroke(
                                        width = 2.dp,
                                        color = if (isActive) Color(0xFF0066CC) else Color(0xFFEAEAEA)
                                    )
                                ) {}
                                if (isActive) {
                                    Box(
                                        modifier = Modifier
                                            .size(10.dp)
                                            .background(Color(0xFF0066CC), CircleShape)
                                    )
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizCountryCodeSheet.swift
import SwiftUI

public struct BizCountryCodeSheetModifier: ViewModifier {
    @Binding var isPresented: Bool
    @Binding var selectedCode: String
    var onChange: ((String) -> Void)?
    
    @State private var searchQuery = ""
    @State private var activeTabIndex = 0
    
    let popularCountries = [
        ("KH", "Cambodia", "+855"),
        ("KR", "Republic of Korea", "+82"),
        ("CN", "China", "+86"),
        ("JP", "Japan", "+81")
    ]
    
    let allCountries = [
        ("KH", "Cambodia", "+855"),
        ("KR", "Republic of Korea", "+82"),
        ("CN", "China", "+86"),
        ("JP", "Japan", "+81"),
        ("US", "United States", "+1"),
        ("GB", "United Kingdom", "+44"),
        ("TH", "Thailand", "+66"),
        ("VN", "Vietnam", "+84"),
        ("SG", "Singapore", "+65")
    ]
    
    var filteredCountries: [(String, String, String)] {
        let list = activeTabIndex == 0 ? popularCountries : allCountries
        if searchQuery.isEmpty { return list }
        let query = searchQuery.lowercased()
        return list.filter {
            $0.1.lowercased().contains(query) || $0.2.contains(query)
        }
    }
    
    public func body(content: Content) -> some View {
        content.bizBottomSheet(
            isPresented: $isPresented,
            content: {
                VStack(alignment: .leading, spacing: 0) {
                    Text("Select Your Country Code")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(Color(hex: "#1A2A5E"))
                        .padding(.bottom, 20)
                    
                    HStack {
                        Image(systemName: "magnifyingglass")
                            .foregroundColor(Color(hex: "#999999"))
                            .font(.system(size: 16))
                        
                        TextField("Search", text: $searchQuery)
                            .font(.system(size: 16))
                            .foregroundColor(Color(hex: "#1A2A5E"))
                    }
                    .padding()
                    .background(Color.clear)
                    .overlay(
                        RoundedRectangle(cornerRadius: 8)
                            .stroke(Color(hex: "#E0E0E0"), lineWidth: 1)
                    )
                    .padding(.bottom, 16)
                    
                    BizSegment(
                        items: ["Popular", "All Country"],
                        selectedIndex: $activeTabIndex
                    )
                    .padding(.bottom, 16)
                    
                    ScrollView {
                        VStack(spacing: 12) {
                            ForEach(filteredCountries, id: \.0) { country in
                                CountryOptionRow(
                                    name: country.1,
                                    dialCode: country.2,
                                    isSelected: selectedCode == country.2,
                                    action: {
                                        selectedCode = country.2
                                        onChange?(country.2)
                                        DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
                                            isPresented = false
                                        }
                                    }
                                )
                            }
                        }
                    }
                    .frame(maxHeight: 350)
                }
                .padding(.horizontal, 24)
                .padding(.top, 8)
                .padding(.bottom, 20)
            }
        )
    }
}

private struct CountryOptionRow: View {
    let name: String
    let dialCode: String
    let isSelected: Bool
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            HStack {
                Text("\(name) (\(dialCode))")
                    .font(.system(size: 14, weight: .medium))
                    .foregroundColor(isSelected ? Color(hex: "#0066CC") : Color(hex: "#111111"))
                
                Spacer()
                
                ZStack {
                    Circle()
                        .stroke(isSelected ? Color(hex: "#0066CC") : Color(hex: "#EAEAEA"), lineWidth: 2)
                        .frame(width: 20, height: 20)
                    
                    if isSelected {
                        Circle()
                            .fill(Color(hex: "#0066CC"))
                            .frame(width: 10, height: 10)
                    }
                }
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 14)
            .background(Color.white)
            .cornerRadius(12)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(isSelected ? Color(hex: "#0066CC") : Color(hex: "#EAEAEA"), lineWidth: 1.5)
            )
        }
    }
}

public extension View {
    func bizCountryCodeSheet(
        isPresented: Binding<Bool>,
        selectedCode: Binding<String>,
        onChange: ((String) -> Void)? = nil
    ) -> some View {
        self.modifier(BizCountryCodeSheetModifier(
            isPresented: isPresented,
            selectedCode: selectedCode,
            onChange: onChange
        ))
    }
}
